### PR TITLE
Change task schema doc to reflect the new implementation

### PR DESCRIPTION
**Pull request by yyscamper**
I had refactored the task schema design (see https://github.com/RackHD/on-tasks/pull/360), but I forgot to update the doc

@anhou @manfrednde @brianparry @iceiilin @pengz1 @lanchongyizu

### CODE REVIEW

File: docs/rackhd/tasks.rst [hunk 1]
```diff
@@ -148,145 +151,88 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
-**Relation between Job & Schema & Task**
+**Schema Classification**
 
-- A Task Definition conforms to only one Task Schema, but different Task Definition can conform to the same Task Schema; For example, all obm-control tasks share the same obm-control schema.
-- A Task Schema only describes one Job, but a Job may be described by multiple Task Schemas. For example, all of install-centos, install-ubuntu, install-esxi schemas are to describe the same install-os job.
-- A Task Schema can include other schemas for the sake of schema reuse and minimizing duplication. For example, all task schemas include a common options schema.
+There are totally 3 kinds of options schema: Common options schema, BaseTask options schema and Task options schema.
 
-Below is a diagram shows the relation:
+- The common options schema is to describe all those common options that shared by all tasks, such as `_taskTimeout`, the common options schema is defined in the file 'https://github.com/RackHD/on-tasks/blob/master/lib/task-data/schemas/common-task-options.json'. User doesn't have to explicitly define the common schema in Task or BaseTask definition, it is defaultly enabled for every task.
```
Comment (pengz1) [inline docs/rackhd/tasks.rst]: that are shared by
defaultly => default

File: docs/rackhd/tasks.rst [hunk 2]
```diff
@@ -148,145 +151,88 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
-describeJob    String  Required  The property is required to point to a job’s di injectable name
-copyright      String  Optional  This attribute shall contain the copyright notice for the schema
-readonly       Boolean Optional  This property shall designate a property to be readonly for user when set to true
-============== ======= ========= =========================================================
-
-**Define a Task Schema**
+- The schema in BaseTask is to describe the options of the corresponding job.
 
-You can follow below 3 steps to define a Task Schema:
+- The schema in Task definition is to describe the options of corresponding task. Since a task defintion will always link to a BaseTask, so the task's schema will automatically inherit the BaseTask's schema during validation. So in practice, usually the task schema only need to describe those options that not included in BaseTask.
```
Comment (pengz1) [inline docs/rackhd/tasks.rst]: need => needs or to remove to
that not included => that are not included or remove that.

File: docs/rackhd/tasks.rst [hunk 3]
```diff
@@ -148,145 +151,88 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
 
-- Step 1: Create a JSON File
+NOTE: The options schema is always optional for Task/BaseTask definition. If missing options schema, that means user gives up the upfront options validation before running a TaskGraph.
```
Comment (pengz1) [inline docs/rackhd/tasks.rst]: If missing options schema => If options schema is missing

File: docs/rackhd/tasks.rst [hunk 4]
```diff
@@ -148,145 +151,88 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
-============== ======= =========================================================
-$schema        String  Specify the version of this schema file, it has to be `rackhd-task-schema.json`
-title          String  Specify a short description for this schema
-description    String  Specify a long & verbose description for this schema
-describeJob    String  Specify the job which this schema is to describe, via job's injectableName
-============== ======= =========================================================
+- Schema File Reference <String>: Specify the file name of a JSON file, the JSON file is a valid JSON schema and it must be placed in the folder 'https://github.com/RackHD/on-tasks/tree/master/lib/task-data/schemas'.
 
-- Step 3: Define Options
+The Built-in Schema is usually used when there is few options or for those situation that is not suitable to use file reference, such as within skupack.
```
Comment (pengz1) [inline docs/rackhd/tasks.rst]: those => the, or just remove those

File: docs/rackhd/tasks.rst [hunk 5]
```diff
@@ -148,145 +151,88 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
+            "commands": {}
+        }
+    }
 
-You can define the default value in the *options* property. These default values will be used as complement if user doesn't pass any value for that option while trigger the task, so all default value should conform to the schema as well.
 
 **Upfront Schema Validation**
 
-The Task Schema validation will be firstly executed when user triggers a workflow. Only if all options (Combine user input and the default value) conform to schema for every task, the workflow then can be successfully triggered.
+The options schema validation will be firstly executed when user triggers a workflow. Only if all options (Combine user input and the default value) conform to all kinds of above schemas for the task, the workflow then can be successfully triggered.
```
Comment (pengz1) [inline docs/rackhd/tasks.rst]: all kinds of => all of, then can => can then

**Comment by yyscamper** [comment]
Thanks @pengz1 , all your comment has been addressed.

**Pull request merged**